e2e/cypress/integration/markdown/markdown_image_spec.js

```diff
@@ -42,4 +42,20 @@ describe('Markdown', () => {
             });
         });
     });
+
+    it('opens preview window when image is clicked', () => {
+        // #  Post markdown message
+        cy.postMessageFromFile('markdown/markdown_inline_images_2.md');
+
+        // Note we use the Gigantic timeout to ensure that the large images will load
+        cy.getLastPostId().then((postId) => {
```

Comment:
On my local machine I needed to set the optional timeout parameter for all the `.then` calls (ex. `.then({timeout: TIMEOUTS.LARGE}, (postId) => {`), without it, tests often failed with a timeout and an error `fn.apply is not a function`. I didn't commit that as I'm not sure if this is only an issue on my local machine.

Reply:
@saturninoabril You would notice that still after my last commit I've not added `wait(TIMEOUTS.SMALL)` to this test - unlike the recent changes from master for the other 2 -, and the test is consistently succeeding without it.

Knowing that waiting for an arbitrary period of time is an anti-pattern, and specially after the fix of `fn.apply` and introducing `waitUntilPermanentPost`, is waiting really needed? why?

Thanks!

Reply:
It's needed at the moment since we don't have extra check that the image is already loaded.  But you're definitely correct that it's an anti-pattern.  Maybe one approach is to add the following?
```javascript
// Add width and height check to verify that img element is actually loaded
cy.get(`#postMessageText_${postId}`).find('img').
    ... // other checks
    and('have.attr', 'alt', 'Github').
    and('have.css', 'height', '38px').
    and('have.css', 'width', '38px');
```
If you agree, above suggestion can be addressed in a separate PR.

Reply:
Sounds good! Let me do it while we are at it, no need to have a separate PR 🙂